src/main/java/com/github/bibsysdev/nva/cristin/person/CristinApiClient.java

```diff
@@ -0,0 +1,190 @@
+        personsWrapper.setFirstRecord(0);
+        personsWrapper.setNextResults(null); // TODO: Change to URI
+        personsWrapper.setHits(transformCristinPersonsToNvaPersons(enrichedPersons));
+        // TODO: NP-2424: Return fields with empty values instead of null to avoid "undefined" in frontend
+        return personsWrapper;
+    }
+
+    /**
+     * Creates a NvaPerson object containing a single transformed CristinPerson. Is used for serialization to the
+     * client.
+     *
+     * @param id       The CristinPersonId of the person to query
+     * @param language Language used for some properties in the Cristin API response
+     * @return a NvaPerson filled with one transformed CristinPerson
+     * @throws BadGatewayException when there is a problem with fetch from backend
+     */
+    public NvaPerson queryOneCristinPerson(String id, String language) throws BadGatewayException {
+        CristinPerson cristinPerson = attemptToGetCristinPerson(id, language);
+        if (cristinPerson == null || !cristinPerson.hasRequiredFields()) {
+            return new EmptyNvaPerson();
+        } else {
+            NvaPerson nvaPerson = new NvaPersonBuilder(cristinPerson).build();
+            nvaPerson.setContext(PERSON_LOOKUP_CONTEXT_URL);
+            return nvaPerson;
+        }
+    }
+
+    protected static <T> T fromJson(InputStreamReader reader, Class<T> classOfT) throws IOException {
+        return OBJECT_MAPPER.readValue(reader, classOfT);
+    }
+
+    // TODO: throw BadGatewayException if this fails as well?
+    protected List<CristinPerson> queryPersons(Map<String, String> parameters) throws IOException,
+                                                                                      URISyntaxException {
+        URL url = generateQueryPersonsUrl(parameters);
+        try (InputStreamReader streamReader = getQueryResponse(url)) {
+            return asList(fromJson(streamReader, CristinPerson[].class));
+        }
+    }
+
+    protected long calculateProcessingTime(long startRequestTime, long endRequestTime) {
+        return endRequestTime - startRequestTime;
+    }
+
+    protected List<CristinPerson> queryAndEnrichPersons(Map<String, String> parameters,
+                                                        String language) throws IOException,
+                                                                                URISyntaxException {
+        List<CristinPerson> persons = queryPersons(parameters);
+        return enrichPersons(language, persons);
+    }
+
+    protected CristinPerson getPerson(String id, String language) throws IOException, URISyntaxException {
+        URL url = generateGetPersonUrl(id, language);
+        try (InputStreamReader streamReader = getResponse(url)) {
+            return fromJson(streamReader, CristinPerson.class);
+        }
+    }
+
+    protected InputStreamReader getQueryResponse(URL url) throws IOException {
+        return new InputStreamReader(url.openStream());
```

Comment:
Better to use here java.net.HttpClient. It is very light but i gives a lot of capabilities in adding headers and also reading easily the response codes and sending a respective error message. I think it will be also easier with handling different encodings. Also, if you pass the httpclient as a field in the class then testing will become much easier.

Reply:
Yes, this is a planned improvement, but might be out of scope for this PR as @torarnet is already doing the same for nva-cristin-project. I believe it would be best if we made a generalized cristinapiclient, seeing as most of the logic (and tests!) are the same.

Reply:
Do we have a task for this?

Reply:
@axthosarouris https://unit.atlassian.net/browse/NP-2439